### PR TITLE
Add BaseDialog Confirm

### PR DESCRIPTION
**Pull request by adrianjost**
# Issue: [VUE-148](https://ticketsystem.schul-cloud.org/browse/VUE-148)

## Description

- refactored BaseDialog to use BaseModal and our new styles
- ~I was unable to write any meaningfull test. I have no idea how to deal with this. If anyone has any idea, please help me!~ Done, except for the auto mount to body.

## Screenshot

image

image

### CODE REVIEW

**Comment by dominikbruechner** [comment]
I committed a test suggestion based on Michael's work... we can discuss more about it once I am back healthy :)

**Comment by adrianjost** [comment]
thanks @dominikbruechner 
I tried debugging these tests and as far as I was able to see these tests don't do anything usfull. At least the HTML is never rendered. We definetly need to dig deeper there, when you are fully back.

**Comment by adrianjost** [comment]
@dominikbruechner I checked out the tests you added again and it turns out they do not render any html what so ever. So these tests are definetly giving us some false confidence.
After the "confirm" method was called the html loooks like: 
`wrapper.element.outerHTML: "<div data-testid="dialog"><!----></div>"`
`wrapper.element.closest("body").outerHTML: "<body><div data-testid="dialog"><!----></div></body>"`
So I am not confident keeping these tests in as they are.

File: src/components/base/BaseDialog/BaseDialog.unit.js [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+
+		const dialogue = wrapper.vm.$dialog.confirm({
+			actionDesign: "danger",
+		});
+		expect(dialogue.isActive).toBe(true);
+		expect(dialogue.currentIconColor).toBe("var(--color-danger)");
+		//TODO Validate more parameters
+		dialogue.close();
+		expect(dialogue.isActive).toBe(false);
+	});
```
Comment (adrianjost) [inline src/components/base/BaseDialog/BaseDialog.unit.js]: These tests don't belong here. They test whether the `index.js` works and therefore they should be in the `index.unit.js` file. They test whether the $dialog property exists in the Vue instance and if the method is doing anything. But they do not test anything from the `BaseDialog.vue` file. So they should not be in this file

**Comment by adrianjost** [comment]
@dominikbruechner I now was able to add a lot of tests. Please review again.

File: src/components/molecules/ModalBodyInfo.vue [hunk 2]
```diff
@@ -3,7 +3,8 @@
 		<div class="icon">
 			<slot name="icon"></slot>
 		</div>
-		<h4>{{ text }}</h4>
+		<!-- eslint-disable-next-line vue/no-v-html -->
+		<h4 v-html="text" />
```
[the remark this answers is not in the document]
Comment (adrianjost) [reply src/components/molecules/ModalBodyInfo.vue]: good point!

[the remark this answers is not in the document]
Comment (dominikbruechner) [reply src/components/molecules/ModalBodyInfo.vue]: maybe even h1 ... not sure... as the content is not selectable any more once the modal is displayed

[the remark this answers is not in the document]
Comment (adrianjost) [reply src/components/molecules/ModalBodyInfo.vue]: good point. Changed it

**Comment by adrianjost** [comment]
@dominikbruechner changed it to an h1 because it's the first heading of this content section => h1 not h2.

**Pull request merged**